### ISSUE TITLE
NPM "php-parser seems to be corrupted" error

### ISSUE THREAD

**Issue by antonioribeiro** [prettier/plugin-php#1813]
Well, sorry, but the issue here is not formatting, we possibly need more types 🤷‍♂️

The problem is that [you are locking this package](https://github.com/prettier/plugin-php/blob/main/package.json#L16) to a specific commit of `php-parser` and Git is having trouble to download a non-corrupt version of it.

It's been impossible to install, as we keep receiving this error message:

```
npm WARN tarball tarball data for php-parser@git+ssh://git@github.com/glayzzle/php-parser.git#e61e26102144f267ecf5e09020865a9baa6ca2f1 (sha512-pEbqx2m7tp+Kgw+/ZrIawbINSml0w/A73veIpFWSequk21nq11qhe7gzpvxOvMfebNbmZX8ohDmPLakOQPGuMA==) 
seems to be corrupted. Trying again.

npm ERR! code EINTEGRITY

npm ERR! sha512-pEbqx2m7tp+Kgw+/ZrIawbINSml0w/A73veIpFWSequk21nq11qhe7gzpvxOvMfebNbmZX8ohDmPLakOQPGuMA== 
integrity checksum failed when using sha512: wanted sha512-pEbqx2m7tp+Kgw+/ZrIawbINSml0w/A73veIpFWSequk21nq11qhe7gzpvxOvMfebNbmZX8ohDmPLakOQPGuMA== 
but got sha512-UAqZS0uEpa6wzDLGw7kfHiJhlMP/dX0HfSqSdqlTcNeuTl9s1VuVuwm/R375XfRm3gYkwfDkfZBJSWSq2EFG/Q==. (144407 bytes)
```

**Comment by czosel** [comment]
Hi @antonioribeiro 

You've closed the issue already - does that mean it has been resolved for you? If so, was the problem somewhere else?

**Comment by antonioribeiro** [comment]
Hey @czosel , yes and no.

The error was related to newer NPM not being compatible with older NPM package-lock.json (6 LTS), apparently. So deleting this file fixes it for us. Still strange that only this package reference was erroring while NPM tried to download it. My "no" is because we had to downgrade our development environments to use 6.14.14. I don't believe this is something you should handle, but it would be nice to not rely on that commit, but I understand how development also works sometimes 😆

Cheers!

**Comment by czosel** [comment]
Interesting - and this happens with the `package-lock.json` in your app, right? (we only have a `yarn.lock`) in this repo)

I hope that we'll soon be able to switch to a released version of the parser, but in the past we often had to make use of direct commit references for quicker turnarounds ...

**Comment by antonioribeiro** [comment]
Yes, if you install an the newer npm, then install packages, then downgrade to npm 6, remove node_modules and do npm install, you will probably be able to reproduce it.

**Comment by nemchik** [comment]
I can confirm this is causing issues for me as well.

This is where the issue comes from https://github.com/glayzzle/php-parser/blob/e61e26102144f267ecf5e09020865a9baa6ca2f1/package-lock.json#L4 (note that the commit in the link is the commit being used as the dependency in this repo as of writing, but their current master branch still has the same line with version 1).

I notice NPM has a version `3.0.3` published 5 months ago, but their github repo does not have a release/tag for it. would using the published version `3.0.3` work here?
